BirthdayGreetingsTest/api.js

```diff
@@ -12,50 +12,45 @@ const app = express();
 const port = 3000;
 app.use(express.json());
 
-app.get("/birthdays/:date", async (req, res) => {
-  // Get all friend objects that have a birthday on the specified date
-  const date = req.params.date;
-  retrieveBirthdaysFromDB(date)
-    .then((friends) => {
-      res.send(friends);
-    })
-    .catch((err) => {
-      res.status(500).send(err);
-    });
+app.get("/birthdays", async (req, res) => {
+  const { startDate, endDate } = req.query;
+  if (startDate && endDate) {
+    // Get all friend objects that have a birthday within the specified range
+    retrieveBirthdaysWithinRangeFromDB(startDate, endDate)
+      .then((friends) => {
+        res.status(200).send(friends);
+      })
+      .catch((err) => {
+        res.status(err.status).send(err);
+      });
+  } else {
+    // Get all friend objects that have a birthday on the specified date
+    const date = req.query.date;
+    retrieveBirthdaysFromDB(date)
+      .then((friends) => {
+        res.status(200).send(friends);
+      })
+      .catch((err) => {
+        res.status(err.status).send(err);
+      });
+  }
 });
 
-app.get("/birthdays/range/:startDate/:endDate", async (req, res) => {
-  // Get all friend objects that have a birthday within the specified range
-  const startDate = req.params.startDate;
-  const endDate = req.params.endDate;
-  retrieveBirthdaysWithinRangeFromDB(startDate, endDate)
-    .then((friends) => {
-      res.send(friends);
-    })
-    .catch((err) => {
-      res.status(500).send(err);
-    });
-});
-
-app.post("/birthdays/send-greeting/:date/:service", (req, res) => {
+app.post("/birthdays/send-greetings", (req, res) => {
```

Comment:
Can you come up with a path which better meets the REST standard?

Reply:
@RichardWatkins1 Something like ` app.post("/birthday-friends/greetings"` ?